src/twisted/web/_newclient.py

```diff
@@ -813,7 +813,14 @@ def writeTo(self, transport):
             else:
                 return self._writeToContentLength(transport)
         else:
-            self._writeHeaders(transport, None)
+            # If the method semantics anticipate a body, include a
```

Comment:
Thanks!  I think `_writeToContentLength` does always write the content length.  The fix is all in `writeTo` which sometimes called `_writeHeaders` directly with no `Content-Length` (or `Transfer-Encoding`).

I guess what's true now is that sometimes a `Content-Length` headers is written by something _other_ tyhan `_writeToContentLength` ... maybe that's what you mean.  And, yea, I can see how that might confuse someone at some point.